pytorch_lightning/tuner/lr_finder.py

```diff
@@ -98,25 +97,16 @@ def __init__(self, mode: str, lr_min: float, lr_max: float, num_training: int):
         self.results = {}
         self._total_batch_idx = 0  # for debug purpose
 
-    def _exchange_scheduler(self, configure_optimizers: Callable):
+    def _exchange_scheduler(self, trainer: "pl.Trainer", model: "pl.LightningModule"):
         """Decorate configure_optimizers methods such that it returns the users originally specified optimizer
         together with a new scheduler that that takes care of the learning rate search."""
+        configure_optimizers = model.configure_optimizers
 
         @wraps(configure_optimizers)
```

Comment:
Your PR changes look good given the existing Tuner design but it builds on top of the existing "hacks" we already have and perhaps we should stop that and start to unravel.

I think that instead of wrapping the `LightningModule` hook to inject the logic, this should patch the functions in `TrainerOptimizersMixin`, otherwise the implementations will continue to diverge

(for example, the tuner does not use `call_hook`)